addons/binding/org.openhab.binding.wso2iots/src/main/java/org/openhab/binding/wso2iots/handler/wso2iotsHandler.java

```diff
@@ -0,0 +1,334 @@
+        if (command instanceof RefreshType) {
+            switch (channelUID.getId()) {
+                case wso2iotsBindingConstants.TEMPERATURE:
+                    updateChannel(channelUID.getId(), iotsResponseTemp);
+                    break;
+                case wso2iotsBindingConstants.LIGHT:
+                    updateChannel(channelUID.getId(), iotsResponseLight);
+                    break;
+                case wso2iotsBindingConstants.MOTION:
+                    updateChannel(channelUID.getId(), iotsResponseMotion);
+                    break;
+                case wso2iotsBindingConstants.HUMIDITY:
+                    updateChannel(channelUID.getId(), iotsResponseHumidity);
+                    break;
+            }
+
+        } else {
+            logger.debug("Can not handle command {}", command);
+        }
+    }
+
+    /**
+     * Start the job refreshing the building monitor data
+     */
+    private void startAutomaticRefresh(wso2iotsConfiguration config) {
+        if (refreshJob == null || refreshJob.isCancelled()) {
+            Runnable runnable = new Runnable() {
+                @Override
+                public void run() {
+                    try {
+                        // Request new data
+                        iotsResponseTemp = updateBuildingMonitorData(wso2iotsBindingConstants.TEMPERATURE);
+                        iotsResponseLight = updateBuildingMonitorData(wso2iotsBindingConstants.LIGHT);
+                        iotsResponseMotion = updateBuildingMonitorData(wso2iotsBindingConstants.MOTION);
+                        iotsResponseHumidity = updateBuildingMonitorData(wso2iotsBindingConstants.HUMIDITY);
+
+                        // Update all channels from the updated data
+                        updateChannel(wso2iotsBindingConstants.TEMPERATURE, iotsResponseTemp);
+                        updateChannel(wso2iotsBindingConstants.LIGHT, iotsResponseLight);
+                        updateChannel(wso2iotsBindingConstants.MOTION, iotsResponseMotion);
+                        updateChannel(wso2iotsBindingConstants.HUMIDITY, iotsResponseHumidity);
+
+                    } catch (Exception e) {
+                        logger.error("Exception occurred during execution: {}", e.getMessage(), e);
+                    }
+                }
+            };
+
+            int delay = config.refresh.intValue();
+            refreshJob = scheduler.scheduleWithFixedDelay(runnable, 0, delay, TimeUnit.MINUTES);
+        }
+    }
+
+    /**
+     * Update the channel from the last building monitor data retrieved
+     *
+     * @param channelId the id identifying the channel to be updated
+     */
+    private void updateChannel(String channelId, BigDecimal iotsResponse) {
+        if (isLinked(channelId)) {
+            State state = null;
+            if (iotsResponse == null) {
+                state = UnDefType.UNDEF;
+            } else if (iotsResponse instanceof BigDecimal) {
```

Comment:
iotsResponse **is** a BigDecimal so why have an else? It will only return false if the value is null. Therefor the else can actually be the test above:
```
if (iotsResponse instanceof BigDecimal) {
  ...
} else {
  state = UnDefType.UNDEF;
}
```